src/js/components/stock-movement-wizard/SendMovementPage.jsx

```diff
@@ -382,9 +383,44 @@ class SendMovementPage extends Component {
     }
   }
 
-  render() {
-    const { previousPage } = this.props;
+  previousPage(values) {
+    if (values.dateShipped && values.shipmentType) {
+      let payload = {
```

Comment:
you are duplicating the code from onSave method, move this to some new method and return this.saveShipment(payload) there so you can specify 'then' and 'catch'